src/tools/emacs.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Emacs
+description: Use Dart on Emacs and its flavors
+toc: false
+---
+
+Emacs has a few major and minor modes built for Dart and Flutter development.
+
+All the steps are identical for enabling Flutter since
+[Dart-Snippets][dart-snippets] contain Flutter snippets too.
+[Flutter.el][flutter-mode] only allows you to run Flutter commands inside your Emacs.
+
+* [Dart-Mode][dart-mode] is the main mode for development.
+* [Dart-Snippets][dart-snippets] enables [Visual Studio Code](/tools/vs-code)
```

Comment:
Since you are mentioning `doom-emacs`, I don't think this mention is really necessary as `doom-emacs` [already has `dart snippets`](https://github.com/hlissner/doom-snippets/tree/master/dart-mode)

Reply:
Yeah, you are right.
But it is there just for the sake of information. For people who use vanilla.
The Snippets in that link are added by very me and their upstream is the Dart-Snippets that I listed in here.

I think it's not necessary to remove these.